GitUI/UserControls/FileStatusDiffCalculator.cs

```diff
@@ -0,0 +1,230 @@
+                fileStatusDescs.AddRange(
+                    revisions
+                        .Where(rev => rev != firstRev && rev != selectedRev)
+                        .Select(rev => new FileStatusWithDescription(
+                            firstRev: rev,
+                            secondRev: selectedRev,
+                            summary: Strings.DiffWithParent + GetDescriptionForRevision(describeRevision, rev.ObjectId),
+                            statuses: module.GetDiffFilesWithSubmodulesStatus(rev.ObjectId, selectedRev.ObjectId, selectedRev.FirstParentId))));
+
+                return fileStatusDescs;
+            }
+
+            // Present common files in BASE->B, BASE->A separately
+            // For the following diff:  A->B a,c,d; BASE->B a,b,c; BASE->A a,b,d
+            // (the file a has unique changes, b has the same change and c,d is changed in one of the branches)
+            // The following groups will be shown: A->B a,c,d; BASE->B a,c; BASE->A a,d; Common BASE b
+            var allBaseToB = module.GetDiffFilesWithSubmodulesStatus(baseRevGuid, selectedRev.ObjectId, selectedRev.FirstParentId);
+            var allBaseToA = module.GetDiffFilesWithSubmodulesStatus(baseRevGuid, firstRev.ObjectId, firstRev.FirstParentId);
+
+            var comparer = new GitItemStatusNameEqualityComparer();
+            var commonBaseToAandB = allBaseToB.Intersect(allBaseToA, comparer).Except(allAToB, comparer).ToList();
+
+            var revBase = new GitRevision(baseRevGuid);
+            fileStatusDescs.Add(new FileStatusWithDescription(
+                firstRev: revBase,
+                secondRev: selectedRev,
+                summary: Strings.DiffBaseToB + GetDescriptionForRevision(describeRevision, selectedRev.ObjectId),
+                statuses: allBaseToB.Except(commonBaseToAandB, comparer).ToList()));
+            fileStatusDescs.Add(new FileStatusWithDescription(
+                firstRev: revBase,
+                secondRev: firstRev,
+                summary: Strings.DiffBaseToB + GetDescriptionForRevision(describeRevision, firstRev.ObjectId),
+                statuses: allBaseToA.Except(commonBaseToAandB, comparer).ToList()));
+            fileStatusDescs.Add(new FileStatusWithDescription(
+                firstRev: revBase,
+                secondRev: selectedRev,
+                summary: Strings.DiffCommonBase + GetDescriptionForRevision(describeRevision, baseRevGuid),
+                statuses: commonBaseToAandB));
+
+            // Add rangeDiff as a separate group (range is not the same as diff with artificial commits)
+            List<GitItemStatus> statuses = new List<GitItemStatus> { new GitItemStatus { Name = Strings.DiffRange, IsRangeDiff = true } };
+
+            var desc = Strings.DiffRange + ": " + GetDescriptionForRevision(describeRevision, firstRevHead) + " -> " +
+                       GetDescriptionForRevision(describeRevision, selectedRevHead);
```

Comment:
Make use of `$"{var}"`?

Reply:
Too late, that was #8471!
To avoid unexpected reviews, I have rebased on master, now that #8471 is merged.
(Could of course change that here too, but I find that the line get too long for $"" to be useful. Could add two temp variables though.)

Reply:
That's why as question not as request. This shouldn't matter because not used extensively. So rather keep it as it is.